src/plugins/network/SphereNetwork/SphereNetwork.cpp

```diff
@@ -65,6 +65,9 @@ bool SphereNetwork::init(std::map<std::string, std::string> &mission_params,
 
     range_ = std::stod(plugin_params.at("range"));
     prob_transmit_ = std::stod(plugin_params.at("prob_transmit"));
+    filter_comms_plane_ = std::stoi(plugin_params.at("filter_comms_plane"));
```

Comment:
Can you use the sc::get<T>() calls instead of std::stoi? (e.g., filter_comms_plane_ = sc::get<bool>("filter_comms_plane", plugin_params, filter_comms_plane_);)